api/apps/api/src/modules/scenarios/locks/lock.service.spec.ts

```diff
@@ -0,0 +1,226 @@
+    await fixtures.GivenNoLockHasBeenGrabbed();
+    await fixtures.ThenOtherUserShouldNotBeAbleToReleaseLock();
+  });
+
+  it(`should not be able to release the lock if lock does not exist`, async () => {
+    await fixtures.GivenNoLockHasBeenGrabbed();
+    await fixtures.ThenShouldNotBeAbleToReleaseLock();
+  });
+
+  it(`isLocked should return true if a lock exists`, async () => {
+    await fixtures.GivenALockExists();
+    await fixtures.ThenIsLockedReturnsTrue();
+  });
+
+  it(`isLocked should return false if lock no lock exists`, async () => {
+    await fixtures.GivenNoLockExists();
+    await fixtures.ThenIsLockedReturnsFalse();
+  });
+});
+
+async function getFixtures() {
+  const fakeUser: User = {
+    id: 'fake-user',
+    email: faker.internet.email(),
+    passwordHash: faker.random.uuid(),
+    isActive: true,
+    isDeleted: false,
+    projects: [],
+    scenarios: [],
+  };
+
+  const fakeScenario: Scenario = {
+    id: faker.random.uuid(),
+    name: 'Scenario',
+    type: ScenarioType.marxan,
+    projectId: 'dummy',
+    status: JobStatus.created,
+    users: [],
+    ranAtLeastOnce: true,
+    createdAt: new Date(),
+    createdByUser: fakeUser,
+    createdBy: 'fake',
+    lastModifiedAt: new Date(),
+  };
+
+  const mockEntityManager = {
+    find: jest.fn(),
+    save: jest.fn(),
+    remove: jest.fn(),
+  };
+
+  const sandbox = await Test.createTestingModule({
+    providers: [
+      LockService,
+      {
+        provide: getRepositoryToken(ScenarioLockEntity),
+        useValue: {
+          manager: {
+            transaction: (fn: (em: Partial<EntityManager>) => Promise<void>) =>
+              fn(mockEntityManager),
+          },
+          find: jest.fn(),
+        },
+      },
+    ],
+  })
+    .compile()
+    .catch((error) => {
+      console.log(error);
+      throw error;
+    });
+
+  const sut = sandbox.get(LockService);
+  const locksRepoMock = sandbox.get(getRepositoryToken(ScenarioLockEntity));
+
+  return {
+    GivenNoLockHasBeenGrabbed: async () => {
+      mockEntityManager.find.mockImplementationOnce(async () => []);
+    },
+    GivenALockHasBeenGrabbed: async () => {
+      mockEntityManager.find.mockImplementationOnce(async () => [
+        {
+          scenarioId: fakeScenario.id,
+          userId: fakeUser.id,
+          createdAt: new Date(),
+        },
+      ]);
+    },
+    GivenNoLockExists: async () => {
+      locksRepoMock.find.mockImplementationOnce(async () => []);
+    },
+    GivenALockExists: async () => {
+      locksRepoMock.find.mockImplementationOnce(async () => [
+        {
+          scenarioId: fakeScenario.id,
+          userId: fakeUser.id,
+          createdAt: new Date(),
+        },
+      ]);
+    },
+    ThenShouldBeAbleToGrabNewLockForGivenScenario: async () => {
+      const mockDate = new Date('2021-02-26T22:42:16.652Z');
+      const spy = jest
+        .spyOn(global, 'Date')
+        .mockImplementation(() => (mockDate as unknown) as string);
```

Comment:
why do we even mock date? as far as I can tell, it isn't used anywhere (but may be should, what do we do about non-released, long-lasting locks?)

Reply:
it will be used in the future (https://vizzuality.atlassian.net/browse/MARXAN-1091)

Reply:
👍 for long-lasting locks, but do we need to mock global Date?

Reply:
sorry, I forgot to comment here, but yeah, we don't need the global mock Date (I removed it in the last commit) :+1: